src/plugins/home/server/services/sample_data/sample_data_installer.ts

```diff
@@ -0,0 +1,205 @@
+    sampleDatasets,
+    logger,
+  }: SampleDataInstallerOptions) {
+    this.esClient = esClient;
+    this.soClient = soClient;
+    this.soImporter = soImporter;
+    this.sampleDatasets = sampleDatasets;
+    this.logger = logger;
+  }
+
+  async install(
+    datasetId: string,
+    installDate: Date = new Date()
+  ): Promise<SampleDataInstallResult> {
+    const sampleDataset = this.sampleDatasets.find(({ id }) => id === datasetId);
+    if (!sampleDataset) {
+      throw new SampleDataInstallError(`Sample dataset ${datasetId} not found`, 404);
+    }
+
+    const nowReference = dateToIso8601IgnoringTime(installDate);
+    const createdDocsPerIndex: Record<string, number> = {};
+
+    for (let i = 0; i < sampleDataset.dataIndices.length; i++) {
+      const dataIndex = sampleDataset.dataIndices[i];
+      const indexName = createIndexName(sampleDataset.id, dataIndex.id);
+      // clean up any old installation of dataset
+      await this.uninstallDataIndex(sampleDataset, dataIndex);
+      await this.installDataIndex(sampleDataset, dataIndex);
+
+      const injectedCount = await insertDataIntoIndex({
+        index: indexName,
+        nowReference,
+        logger: this.logger,
+        esClient: this.esClient,
+        dataIndexConfig: dataIndex,
+      });
+      createdDocsPerIndex[indexName] = injectedCount;
+    }
+
+    const createdSavedObjects = await this.importSavedObjects(sampleDataset);
+
+    return {
+      createdDocsPerIndex,
+      createdSavedObjects,
+    };
+  }
+
+  async uninstall(datasetId: string) {
+    const sampleDataset = this.sampleDatasets.find(({ id }) => id === datasetId);
+    if (!sampleDataset) {
+      throw new SampleDataInstallError(`Sample dataset ${datasetId} not found`, 404);
+    }
+
+    for (let i = 0; i < sampleDataset.dataIndices.length; i++) {
+      const dataIndex = sampleDataset.dataIndices[i];
+      await this.uninstallDataIndex(sampleDataset, dataIndex);
+    }
+    const deletedObjects = await this.deleteSavedObjects(sampleDataset);
+
+    return {
+      deletedSavedObjects: deletedObjects,
+    };
+  }
+
+  private async uninstallDataIndex(dataset: SampleDatasetSchema, dataIndex: DataIndexSchema) {
+    let index = createIndexName(dataset.id, dataIndex.id);
+
+    try {
+      // if the sample data was reindexed using UA, the index name is actually an alias pointing to the reindexed
+      // index. In that case, we need to get rid of the alias and to delete the underlying index
+      const { body: response } = await this.esClient.asCurrentUser.indices.getAlias({
+        name: index,
+      });
+      const aliasName = index;
+      index = Object.keys(response)[0];
+      await this.esClient.asCurrentUser.indices.deleteAlias({ name: aliasName, index });
+    } catch (err) {
+      // ignore errors from missing alias
+    }
+
+    try {
+      await this.esClient.asCurrentUser.indices.delete({
+        index,
+      });
+    } catch (err) {
+      // ignore delete errors
+    }
```

Comment:
This is the actual fix, and the only functional change of the PR. We're now checking if there's an alias matching the index name, and if there is, we delete both the alias and the underlying index